<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from test_robotiq_2f_85.xacro       | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- 
#####################################################################################
#
# Copyright (c) 2023-2025 Galbot, Inc. All Rights Reserved.
#
# This software contains confidential and proprietary information of Galbot, Inc.
# ("Confidential Information"). You shall not disclose such Confidential Information
# and shall use it only in accordance with the terms of the license agreement you
# entered into with Galbot, Inc.
#
# UNAUTHORIZED COPYING, USE, OR DISTRIBUTION OF THIS SOFTWARE, OR ANY PORTION OR
# DERIVATIVE THEREOF, IS STRICTLY PROHIBITED. IF YOU HAVE RECEIVED THIS SOFTWARE IN
# ERROR, PLEASE NOTIFY GALBOT, INC. IMMEDIATELY AND DELETE IT FROM YOUR SYSTEM.
#
#####################################################################################
#
# Description: Xacro file for testing robotiq gripper.
# Author: Herman Ye@Galbot
#
#####################################################################################
-->
<robot name="test_robotiq_2f_85">
  <!-- Parent link -->
  <link name="test_base_link"/>
  <link name="left_gripper_camera_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/d405_camera_link.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_0.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_1.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_2.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_3.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_4.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_5.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_6.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_7.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_8.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_9.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_10.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_11.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_12.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_13.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_14.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_15.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_16.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_17.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_18.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_19.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_20.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_21.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_22.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_23.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_24.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_25.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_26.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_27.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_28.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_29.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_30.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_31.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/d405_camera_link_convex_decomposition/d405_camera_link_collision_32.obj"/>
      </geometry>
    </collision>
  </link>
  <link name="left_gripper_galbot_to_robotiq_adapter_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-1.9028E-07 5.9618E-18 -0.0041416"/>
      <mass value="0.023056"/>
      <inertia ixx="5.5937E-06" ixy="-4.3322E-08" ixz="2.321E-10" iyy="5.3172E-06" iyz="1.9904E-12" izz="1.0645E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/galbot_to_robotiq_adapter.obj"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/galbot_to_robotiq_adapter_convex.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="left_gripper_robotiq_85_base_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/robotiq_base.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/robotiq_base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 2.274e-05 0.03232288"/>
      <mass value="6.6320197e-01"/>
      <inertia ixx="5.1617816e-04" ixy="2.936e-8" ixz="0.0" iyy="5.8802208e-04" iyz="-3.2296e-7" izz="3.9462776e-04"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_left_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/left_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01213197 0.0002 -0.00058647"/>
      <mass value="1.384773208e-02"/>
      <inertia ixx="3.5232e-7" ixy="0.0" ixz="1.1744e-7" iyy="2.31944e-6" iyz="0" izz="2.23136e-6"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_right_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/right_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01213197 0.0002 -0.00058647"/>
      <mass value="1.384773208e-02"/>
      <inertia ixx="3.5232e-7" ixy="0.0" ixz="-1.1744e-7" iyy="2.31944e-6" iyz="0.0" izz="2.23136e-6"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_left_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/left_finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00346899 -0.00079447 0.01867121"/>
      <mass value="4.260376752e-02"/>
      <inertia ixx="1.385792000000000e-05" ixy="0.0" ixz="-2.17264e-06" iyy="1.183208e-05" iyz="0.0" izz="5.19672e-06"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_right_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/right_finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00346899 -5.53e-06 0.01867121"/>
      <mass value="4.260376752000000e-02"/>
      <inertia ixx="1.385792e-05" ixy="0.0" ixz="2.17264e-06" iyy="1.183208e-05" iyz="0.0" izz="5.19672e-06"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_left_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/left_inner_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_inner_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01897699 0.00015001 0.02247101"/>
      <mass value="2.969376448e-02"/>
      <inertia ixx="9.57136e-06" ixy="0.0" ixz="-3.93424e-06" iyy="8.69056e-06" iyz="0.0" izz="8.19144e-06"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_right_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/right_inner_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_inner_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01926824 5.001e-05 0.02222178"/>
      <mass value="2.969376448e-02"/>
      <inertia ixx="9.42456e-06" ixy="0.0" ixz="3.9636e-06" iyy="8.69056e-06" iyz="0.0" izz="8.33824e-06"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_left_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/left_finger_tip.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_0.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_1.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_2.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_3.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_4.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_5.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_6.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_7.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_8.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/left_finger_tip_convex_decomposition/left_finger_tip_collision_9.obj"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01456706 -0.0008 0.01649701"/>
      <mass value="4.268588744e-02"/>
      <inertia ixx="1.048152e-05" ixy="0.0" ixz="3.5232e-6" iyy="1.197888e-05" iyz="0.0" izz="4.22784e-06"/>
    </inertial>
  </link>
  <link name="left_gripper_robotiq_85_right_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/visual/2f_85/right_finger_tip.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_0.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_1.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_2.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_3.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_4.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_5.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_6.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_7.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_8.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_9.obj"/>
      </geometry>
    </collision>
    <collision>
      <geometry>
        <mesh filename="package://galbot_one_charlie_description/meshes/gripper/robotiq_gripper/collision/2f_85/right_finger_tip_convex_decomposition/right_finger_tip_collision_10.obj"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01456706 5e-05 0.01649701"/>
      <mass value="4.268588744e-02"/>
      <inertia ixx="1.048152e-05" ixy="0.0" ixz="-3.5232e-06" iyy="1.197888e-05" iyz="0.0" izz="4.22784e-06"/>
    </inertial>
  </link>
  <joint name="left_gripper_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="test_base_link"/>
    <child link="left_gripper_galbot_to_robotiq_adapter_link"/>
  </joint>
  <joint name="left_gripper_camera_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="test_base_link"/>
    <child link="left_gripper_camera_link"/>
  </joint>
  <joint name="left_gripper_galbot_to_robotiq_adapter_joint" type="fixed">
    <parent link="left_gripper_galbot_to_robotiq_adapter_link"/>
    <child link="left_gripper_robotiq_85_base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="left_gripper_robotiq_85_left_knuckle_joint" type="revolute">
    <parent link="left_gripper_robotiq_85_base_link"/>
    <child link="left_gripper_robotiq_85_left_knuckle_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.03060114 0.0 0.05490452"/>
    <limit effort="50" lower="-0.8" upper="0" velocity="0.5"/>
    <mimic joint="left_gripper_robotiq_85_right_knuckle_joint" multiplier="-1"/>
  </joint>
  <joint name="left_gripper_robotiq_85_right_knuckle_joint" type="revolute">
    <parent link="left_gripper_robotiq_85_base_link"/>
    <child link="left_gripper_robotiq_85_right_knuckle_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.03060114 0.0 0.05490452"/>
    <limit effort="50" lower="0" upper="0.8" velocity="0.5"/>
  </joint>
  <joint name="left_gripper_robotiq_85_left_finger_joint" type="fixed">
    <parent link="left_gripper_robotiq_85_left_knuckle_link"/>
    <child link="left_gripper_robotiq_85_left_finger_link"/>
    <origin rpy="0 0 0" xyz="0.03152616 0.0 -0.00376347"/>
  </joint>
  <joint name="left_gripper_robotiq_85_right_finger_joint" type="fixed">
    <parent link="left_gripper_robotiq_85_right_knuckle_link"/>
    <child link="left_gripper_robotiq_85_right_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.03152616 0.0 -0.00376347"/>
  </joint>
  <joint name="left_gripper_robotiq_85_left_inner_knuckle_joint" type="continuous">
    <parent link="left_gripper_robotiq_85_base_link"/>
    <child link="left_gripper_robotiq_85_left_inner_knuckle_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.0127 0.0 0.06142"/>
    <limit effort="50" lower="-0.8" upper="0" velocity="0.5"/>
    <mimic joint="left_gripper_robotiq_85_right_knuckle_joint" multiplier="-1"/>
  </joint>
  <joint name="left_gripper_robotiq_85_right_inner_knuckle_joint" type="continuous">
    <parent link="left_gripper_robotiq_85_base_link"/>
    <child link="left_gripper_robotiq_85_right_inner_knuckle_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.0127 0.0 0.06142"/>
    <limit effort="50" lower="0" upper="0.8" velocity="0.5"/>
    <mimic joint="left_gripper_robotiq_85_right_knuckle_joint" multiplier="1"/>
  </joint>
  <joint name="left_gripper_robotiq_85_left_finger_tip_joint" type="continuous">
    <parent link="left_gripper_robotiq_85_left_finger_link"/>
    <child link="left_gripper_robotiq_85_left_finger_tip_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.00563134 0.0 0.04718515"/>
    <limit effort="50" lower="0" upper="0.8" velocity="0.5"/>
    <mimic joint="left_gripper_robotiq_85_right_knuckle_joint" multiplier="1"/>
  </joint>
  <joint name="left_gripper_robotiq_85_right_finger_tip_joint" type="continuous">
    <parent link="left_gripper_robotiq_85_right_finger_link"/>
    <child link="left_gripper_robotiq_85_right_finger_tip_link"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.00563134 0.0 0.04718515"/>
    <limit effort="50" lower="-0.8" upper="0" velocity="0.5"/>
    <mimic joint="left_gripper_robotiq_85_right_knuckle_joint" multiplier="-1"/>
  </joint>
  <!-- For tcp-->
  <link name="left_gripper_tcp_link"/>
  <joint name="left_gripper_tcp_joint" type="fixed">
    <parent link="left_gripper_robotiq_85_base_link"/>
    <child link="left_gripper_tcp_link"/>
    <origin rpy="-1.5707963267948966 -1.5707963267948966 0" xyz="0.0 0.0 0.15"/>
    <!-- 0.17084 is the distance between the gripper base link and the tcp link -->
  </joint>
</robot>
